File: src/python/finished/1_bit_and_2_bit_characters.py
# https://leetcode.com/problems/1-bit-and-2-bit-characters/

from typing import List


class Solution:
    def isOneBitCharacter(self, bits: List[int]) -> bool:
        n = len(bits)
        i = 0
        while i < (n - 1):
            if bits[i] == 0:
                i += 1
            elif bits[i] == 1:
                i += 2

        return i == n - 1

File: src/rust/finished/1_bit_and_2_bit_characters.rs
impl Solution {
    pub fn is_one_bit_character(bits: Vec<i32>) -> bool {
        let mut i = 0_usize;
        let n = bits.len();
        while i < n - 1 {
            match bits[i] {
                0 => i += 1,
                1 => i += 2,
                _ => unreachable!(),
            }
        }
        i == n - 1
    }
}
